app/controllers/spree/admin/reports_controller.rb

```diff
@@ -217,7 +217,11 @@ def render_content?
       end
 
       def render_report(header, table, create_csv, csv_file_name)
-        send_data csv_report(header, table), filename: csv_file_name if create_csv
+        if create_csv
+          @csv_report = csv_report(header, table)
+          send_data @csv_report, filename: csv_file_name
+        end
```

Comment:
from "rapper style" code to "daily language" code :heart:

Reply:
Note however that my whole point was to introduce an ivar (which I absolutely dislike) so I could test the csv output. I wanted to avoid any test mocks if possible.

Reply:
ah, I see, much better anyway!